gems/pending/Scvmm/miq_hyperv_disk.rb

```diff
@@ -0,0 +1,171 @@
+$LOAD_PATH.push("#{File.dirname(__FILE__)}/../util")
+
+require 'miq_winrm'
+require 'miq_scvmm_parse_powershell'
+require 'base64'
+require 'securerandom'
+require 'memory_buffer'
+
+require 'rufus/lru'
+
+class MiqHyperVDisk
+  MIN_SECTORS_TO_CACHE = 8
+  DEF_BLOCK_CACHE_SIZE = 300
+  DEBUG_CACHE_STATS    = false
+
+  attr_reader :hostname, :virtual_disk, :file_offset, :file_size, :parser, :vm_name, :temp_snapshot_name
+
+  def initialize(hyperv_host, user, pass, port = nil)
+    @hostname  = hyperv_host
+    @winrm     = MiqWinRM.new
+    port ||= 5985
+    options = {:port => port, :user => user, :pass => pass, :hostname => @hostname}
+    @connection  = @winrm.connect(options)
+    @parser      = MiqScvmmParsePowershell.new
+    @block_size  = 4096
+    @file_size   = 0
+    @block_cache = LruHash.new(DEF_BLOCK_CACHE_SIZE)
+    @cache_hits  = {}
+    @cache_misses = {}
+    @total_read_execution_time = @total_copy_from_remote_time = 0
+  end
+
+  def open(vm_disk)
+    @virtual_disk  = vm_disk
+    @file_offset   = 0
+    stat_script = <<-STAT_EOL
+    (Get-Item "#{@virtual_disk}").length
+STAT_EOL
+    file_size, _stderr   = @parser.parse_single_powershell_value(@winrm.run_powershell_script(stat_script))
+    @file_size           = file_size.to_i
+    @end_byte_addr       = @file_size - 1
+    @size_in_blocks, rem = @file_size.divmod(@block_size)
+    @size_in_blocks += 1 if rem > 0
+    (0..@size_in_blocks - 1).each do |block|
+      @cache_hits[block]   = 0
+      @cache_misses[block] = 0
+    end
```

Comment:
This initialization block can be removed, given the changes described in `initialize`.